ignite/services/network/publish.go

```diff
@@ -14,16 +18,54 @@ import (
 	"github.com/ignite-hq/cli/ignite/services/network/networktypes"
 )
 
+// SharePercentage represent percent of total share
+type SharePercentage struct {
+	denom   string
+	percent float64
+}
+
+// NewSharePercentage creates new share percentage
+func NewSharePercentage(denom string, percent float64) SharePercentage {
+	return SharePercentage{denom: denom, percent: percent}
+}
+
+// ParseSharePercentages parsers SharePercentage list from string
+// format: 12.4%foo,10%bar,0.133%baz
+func ParseSharePercentages(percentagesString string) ([]SharePercentage, error) {
```

Comment:
The definition might look a bit cleaner by using `percentages` instead of `percentagesString`. The function is defining a var with the same name internally, but it would be fine if the internal var would be called `ps` (or similar).